src/AudioManagerImpl.js

```diff
@@ -221,18 +222,25 @@ export default class AudioManagerImpl implements AudioManager {
         });
     }
 
-    playAnnouncement(messageIdSuffix: string, intl: IntlShape, messagePayload: any) {
-        if (this.announcementsEnabled) {
-            if (window.speechSynthesis.speaking || window.speechSynthesis.pending) {
-                window.speechSynthesis.cancel();
-            }
-            const messageId = "Announcement." + messageIdSuffix;
-            const toAnnounce = intl.formatMessage({ id: messageId}, messagePayload);
-            const utterance = new SpeechSynthesisUtterance(toAnnounce);
-            window.speechSynthesis.speak(utterance);
+    playFeedbackAnnouncement(message: string) {
+        if (this.audioFeedbackEnabled) {
+            this.cancelSpeech();
+            this.playStringMessage(message);
+        }
+    }
+
+    playPreviewAnnouncement(message: string) {
+        if (this.audioPreviewEnabled) {
+            this.cancelSpeech();
+            this.playStringMessage(message);
         }
     }
 
+    playStringMessage(message: string) {
+        const utterance = new SpeechSynthesisUtterance(message);
+        window.speechSynthesis.speak(utterance);
```

Comment:
In the main conversation I commented on cases in which we'd want both feedback (state change) and preview (focus change) announcements.  Given that these are independently triggered, one option is to make some kind of queue for utterances and regularly check to see if we're still speaking or preparing to speak.  If we're no longer speaking and there are still unplayed utterances, pull the next one out of the queue.

Reply:
That would cause a huge delay in the announcements when users are rapidly tabbing through the components. I think it makes sense to play one announcement at a time, regardless of the announcement type.